src/client/ContentClient.ts

```diff
@@ -0,0 +1,223 @@
+    const areWeRunningInNode = isNode()
+
+    const form = new FormData()
+    form.append('entityId', deployData.entityId)
+    addModelToFormData(deployData.authChain, form, 'authChain')
+
+    const alreadyUploadedHashes = await hashesAlreadyOnServer(Array.from(deployData.files.keys()), options)
+    for (const [fileHash, file] of deployData.files) {
+      if (!alreadyUploadedHashes.has(fileHash) || fileHash === deployData.entityId) {
+        if (areWeRunningInNode) {
+          // Node
+          form.append(fileHash, Buffer.isBuffer(file) ? file : Buffer.from(arrayBufferFrom(file)), fileHash)
+        } else {
+          // Browser
+          form.append(fileHash, new Blob([arrayBufferFrom(file)]), fileHash)
+        }
+      }
+    }
+
+    return form
+  }
+
+  async function deploy(deployData: DeploymentData, options?: RequestOptions): Promise<unknown> {
+    const form = await buildEntityFormDataForDeployment(deployData, options)
+
+    const requestOptions = mergeRequestOptions(options ? options : {}, {
+      body: form as any,
+      method: 'POST'
+    })
+
+    return await fetcher.fetch(`${contentUrl}/entities`, requestOptions)
+  }
+
+  async function fetchEntitiesByPointers(pointers: string[], options?: RequestOptions): Promise<Entity[]> {
+    if (pointers.length === 0) {
+      return Promise.reject(`You must set at least one pointer.`)
+    }
+
+    const requestOptions = mergeRequestOptions(options ? options : {}, {
+      body: JSON.stringify({ pointers }),
+      method: 'POST',
+      headers: { 'Content-Type': 'application/json' }
+    })
+
+    return (await fetcher.fetch(`${contentUrl}/entities/active`, requestOptions)).json()
+  }
+
+  async function fetchEntitiesByIds(ids: string[], options?: RequestOptions): Promise<Entity[]> {
+    if (ids.length === 0) {
+      return Promise.reject(`You must set at least one id.`)
+    }
+
+    const requestOptions = mergeRequestOptions(options ? options : {}, {
+      body: JSON.stringify({ ids }),
+      method: 'POST',
+      headers: { 'Content-Type': 'application/json' }
+    })
+
+    return (await fetcher.fetch(`${contentUrl}/entities/active`, requestOptions)).json()
+  }
+
+  async function fetchEntityById(id: string, options?: RequestOptions): Promise<Entity> {
+    const entities: Entity[] = await fetchEntitiesByIds([id], options)
+    if (entities.length === 0) {
+      return Promise.reject(`Failed to find an entity with id '${id}'.`)
+    }
+    return entities[0]
+  }
+
+  async function downloadContent(contentHash: string, options?: Partial<RequestOptions>): Promise<Buffer> {
+    const { attempts = 3, waitTime = 500 } = options ? options : {}
+    const timeout = options?.timeout ? { timeout: options.timeout } : {}
+
+    return retry(
+      `fetch file with hash ${contentHash} from ${contentUrl}`,
+      async () => {
+        const content = await (await fetcher.fetch(`${contentUrl}/contents/${contentHash}`, timeout)).buffer()
```

Comment:
Can you make this one `${baseUrl}/${hash}`? To enable downloads from any static server like sdk7-goerli-plaza

Reply:
not quite sure to follow, from where you would take the baseUrl?

Reply:
If you are using a content server it is `https://peer.decentraland.org/content/contents`

If you are instead using a worlds content server, it comes as part of the URN like this

`urn:decentraland:entity:bafkreihiv5zkzjui46gvxtsnk5pfogmq7kyzijxpf3gjqlb2ivydcuwgxq?baseUrl=https://worlds-content-server.decentraland.org/contents/` (menduz.dcl.eth)

Or in the case of the goerli plaza `urn:decentraland:entity:bafkreiavggsbbao4hqz4mizdcvcouqnkmjiw4hp33gynrbggq73jjmvzt4?=&baseUrl=https://sdk-team-cdn.decentraland.org/ipfs/` (from https://sdk-team-cdn.decentraland.org/ipfs/goerli-plaza-main/about)

The baseUrl for the _content_ is different than the baseUrl of the _content-server_

More info here https://adr.decentraland.org/adr/ADR-207

Reply:
sorry if I'm been a bit slow but it seems to me the cases you are mentioning are covered, you can call `createContentClient` with `https://sdk-team-cdn.decentraland.org/ipfs/` or `https://worlds-content-server.decentraland.org/`, are you perhaps talking about how the catalyst client should create different clients by parsing the urns?